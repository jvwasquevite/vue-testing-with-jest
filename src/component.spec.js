import Vue from "vue/dist/vue.common"
import Component from "./component"

describe("Shopping list component", () => {
  let $mounted

  beforeEach(() => {
    $mounted = new Vue(Component).$mount()
  })

  /**
   * HTML DOM VERIFICATION
   *
   * Using different snapshots to check
   * and test the output of rendered HTML
   */

  test("should make shopping-list component match snapshot dom", () => {
    let $html = $mounted.$el.outerHTML

    /**
     * Checks for component snapshot #1
     */

    expect($html).toMatchSnapshot()

    /**
     * Checks if the first li class
     * element is set as 'done'
     */

    $mounted.items[1].done = true

    /**
     * Checks for component snapshot #2
     */

    Vue.nextTick(() => {
      $html = $mounted.$el.outerHTML
      expect($html).toMatchSnapshot()
    })
  })

  /**
   * TESTING EVENTS ON HTML DOM
   *
   * Verifying if the click event is
   * correctly set on the button elements
   *
   * Testing if the click event is calling the complete() method,
   * to assert the result of that call at the same time
   */

  test("should check click event and set it to true", () => {
    let lis = $mounted.$el.querySelectorAll("li")
    let button = lis[1].querySelector("button")

    let customEvent = new Event("click")
    button.dispatchEvent(customEvent)

    expect($mounted.items[1].done).toBeTruthy()
  })
})
