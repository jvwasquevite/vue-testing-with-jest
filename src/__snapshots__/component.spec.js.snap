// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shopping list component should make shopping-list component match snapshot dom 1`] = `
"<ul class=\\"shopping-list\\"><li class=\\"done\\">
        Beer
        <button>Done</button></li><li class=\\"\\">
        Milk
        <button>Done</button></li><li class=\\"\\">
        Apple
        <button>Done</button></li></ul>"
`;

exports[`Shopping list component should make shopping-list component match snapshot dom 2`] = `
"<ul class=\\"shopping-list\\"><li class=\\"done\\">
        Beer
        <button>Done</button></li><li class=\\"done\\">
        Milk
        <button>Done</button></li><li class=\\"\\">
        Apple
        <button>Done</button></li></ul>"
`;
